### DISCUSSION TITLE
Return value of autogenerated Halide function

### DISCUSSION THREAD

**Discussion by rteja** [halide/Halide#6241] [Q&A] [answered]
When Halide Generator is used to auto-generate the C function object containing the halide function, can we always assume that the prototype of the function that is generated always has a return type as integer. I could not find anything related to this in Halide documentation pages.

**Comment by ashishUthama** [comment, marked as answer]
Based on [CodeGen_C](https://github.com/halide/Halide/blob/f43f01635f91a3361045295bf48f2becac2f54f7/src/CodeGen_C.cpp#L1495)
```

CodeGen_C.cpp:1495:        stream << "\nHALIDE_FUNCTION_ATTRS\nint " << function_name << "_argv(void **args);\n"; 
CodeGen_C.cpp:1499:    stream << "\nHALIDE_FUNCTION_ATTRS\nint " << function_name << "_argv(void **args) {\n";
```

I would say 'yes'.

**Comment by abadams** [comment]
Yes, you can assume that. The error codes that can be returned by a Halide pipeline are listed here: 
https://github.com/halide/Halide/blob/master/src/runtime/HalideRuntime.h#L990

**Reply by stewpend0us** [reply]
I think this is now here: [https://github.com/halide/Halide/blob/master/src/runtime/HalideRuntime.h#L1069](url)

maybe search the file for ```enum halide_error_code_t``` in the future?
